osu.Game/Rulesets/Ruleset.cs

```diff
@@ -33,6 +34,8 @@ public IEnumerable<Mod> GetAllMods() => Enum.GetValues(typeof(ModType)).Cast<Mod
 
         public abstract IEnumerable<Mod> GetModsFor(ModType type);
 
+        public abstract IEnumerable<Mod> GetLegacyModsFor(LegacyMods mods);
```

Comment:
* Virtual, not abstract.
* Rename to `ConvertLegacyMods`.
* xmldoc something like `Converts mods from legacy enum values. Do not override if you're not a legacy ruleset.`